enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index])
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    while index < size
+      yield(self[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each do |element|
+      result = yield(element)
+      next unless result
+
+      selected << element
+    end
+    selected
+  end
+
+  def my_all?
+    if block_given?
+      my_each { |element| return false if yield(element) == false }
+    else
+      my_each { |element| return false if element.nil? || element == false }
+    end
+    true
+  end
+
+  def my_any?
+    if block_given?
+      my_each { |element| return true if yield(element) }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
+
+  def my_none?
+    if block_given?
+      my_each { |element| return false if yield(element) }
+    else
+      my_each { |element| return false if element }
+    end
+    true
+  end
```

Comment:
Please, see comment on `my_all?`

### Find more details about [`none?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F)

[OPTIONAL]

I think if we could get the block from `my_none?` to `my_any?` we should only need something like `!my_any?` for `my_none?` to work fine.

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use `block.call(element)` instead of `yield(element)` in `my_any?`